tests/e2e/util/kubectl/kubectl.go

```diff
@@ -129,18 +129,31 @@ func (k Kubectl) ApplyString(yamlString string) error {
 
 // Apply applies the given yaml file to the cluster
 func (k Kubectl) Apply(yamlFile string) error {
-	err := k.ApplyWithLabels(yamlFile, "")
-	return err
+	return k.applyWithOptions("-f " + yamlFile)
 }
 
 // ApplyWithLabels applies the given yaml file to the cluster with the given labels
 func (k Kubectl) ApplyWithLabels(yamlFile, label string) error {
-	cmd := k.build(" apply " + labelFlag(label) + " -f " + yamlFile)
+	return k.applyWithOptions(labelFlag(label) + " -f " + yamlFile)
+}
+
+// ApplyKustomize applies the given kustomization file to the cluster
+func (k Kubectl) ApplyKustomize(kustomizationDirectory string) error {
+	return k.applyWithOptions("-k " + kustomizationDirectory)
+}
+
+// ApplyKustomizeWithLabels applies the given kustomization file to the cluster with the given labels
+func (k Kubectl) ApplyKustomizeWithLabels(kustomizationDirectory, label string) error {
+	return k.applyWithOptions(labelFlag(label) + " -k " + kustomizationDirectory)
+}
```

Comment:
```suggestion
func (k Kubectl) ApplyKustomize(appName string, labels ...string) error {
	cmd := "-k " + common.GetKustomizeDir(appName)
	for _, label := range labels {
		cmd += labelFlag(label)
	}
	return k.applyWithOptions(cmd)
}
```

This can be simplified into a single function to handle both cases.